src/accelerate/utils/modeling.py

```diff
@@ -856,6 +856,19 @@ def get_balanced_memory(
     return max_memory
 
 
+def calculate_maximum_sizes(model: torch.nn.Module):
+    "Computes the total size of the model and its largest layer"
+    sizes = compute_module_sizes(model)
+    modules_to_treat = (
+        list(model.named_parameters(recurse=False))
+        + list(model.named_children())
+        + list(model.named_buffers(recurse=False))
+    )
+    largest_layer = get_max_layer_size(modules_to_treat, sizes, [])
```

Comment:
The `no_split_module_classes` is what will determine the size of the largest layer very often, so this info should be passed along if we have it (from a Transformers model for instance).

Reply:
Done.